stackslib/src/net/api/tests/postblock_proposal.rs

```diff
@@ -14,48 +14,38 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::cell::RefCell;
+use std::collections::VecDeque;
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
-use std::rc::Rc;
 use std::sync::{Arc, Condvar, Mutex};
 
-use clarity::types::chainstate::{StacksPrivateKey, TrieHash};
-use clarity::util::secp256k1::MessageSignature;
-use clarity::util::vrf::VRFProof;
+use clarity::consts::CHAIN_ID_TESTNET;
+use clarity::types::chainstate::StacksPrivateKey;
 use clarity::vm::ast::ASTRules;
-use clarity::vm::types::{PrincipalData, QualifiedContractIdentifier, StacksAddressExtensions};
-use clarity::vm::{ClarityName, ContractName, Value};
+use clarity::vm::types::StandardPrincipalData;
 use mempool::{MemPoolDB, MemPoolEventDispatcher, ProposalCallbackReceiver};
 use postblock_proposal::{NakamotoBlockProposal, ValidateRejectCode};
-use stacks_common::bitvec::BitVec;
-use stacks_common::types::chainstate::{ConsensusHash, StacksAddress};
-use stacks_common::types::net::PeerHost;
-use stacks_common::types::{Address, StacksEpochId};
-use stacks_common::util::hash::{hex_bytes, Hash160, MerkleTree, Sha512Trunc256Sum};
+use stacks_common::types::chainstate::ConsensusHash;
+use stacks_common::types::StacksEpochId;
 
 use super::TestRPC;
 use crate::chainstate::burn::db::sortdb::SortitionDB;
-use crate::chainstate::burn::BlockSnapshot;
 use crate::chainstate::nakamoto::miner::NakamotoBlockBuilder;
-use crate::chainstate::nakamoto::{NakamotoBlock, NakamotoBlockHeader, NakamotoChainState};
+use crate::chainstate::nakamoto::NakamotoChainState;
 use crate::chainstate::stacks::db::StacksChainState;
 use crate::chainstate::stacks::miner::{BlockBuilder, BlockLimitFunction};
-use crate::chainstate::stacks::test::{make_codec_test_block, make_codec_test_nakamoto_block};
-use crate::chainstate::stacks::{
-    CoinbasePayload, StacksBlockHeader, StacksTransactionSigner, TenureChangeCause,
-    TenureChangePayload, TokenTransferMemo, TransactionAnchorMode, TransactionAuth,
-    TransactionPayload, TransactionPostConditionMode, TransactionVersion,
+use crate::chainstate::stacks::test::make_codec_test_nakamoto_block;
+use crate::core::test_util::{make_stacks_transfer_tx, to_addr};
+use crate::net::api::postblock_proposal::{
+    BlockValidateOk, BlockValidateReject, TEST_REPLAY_TRANSACTIONS,
 };
-use crate::core::BLOCK_LIMIT_MAINNET_21;
 use crate::net::api::*;
 use crate::net::connection::ConnectionOptions;
-use crate::net::httpcore::{
-    HttpRequestContentsExtensions, RPCRequestHandler, StacksHttp, StacksHttpRequest,
-};
+use crate::net::httpcore::{RPCRequestHandler, StacksHttp, StacksHttpRequest};
 use crate::net::relay::Relayer;
 use crate::net::test::TestEventObserver;
-use crate::net::{ProtocolFamily, TipRequest};
+use crate::net::ProtocolFamily;
 
+#[warn(unused)]
```

Comment:
Nothing you need to change in this PR, but I forgot that warning on unused was still disabled. It seems worth doing a pass to remove some of those allows that apply to the entire library.